crates/rpc-types-engine/src/attributes.rs

```diff
@@ -56,55 +55,80 @@ impl OpPayloadAttributes {
+                .try_into_recovered()
+                .map_err(|_| SignatureError::FromBytes("Failed to recover signature"))?;
+
+            let recovered_op_tx = OpTxEnvelope::try_from_eth_envelope(recovered.inner().clone())
+                .map_err(|_| {
+                    SignatureError::FromBytes("Failed to convert back from eth envelope")
+                })?;
+
+            Ok(Recovered::new_unchecked(recovered_op_tx, recovered.signer()))
+        })
     }
 
-    /// Returns an iterator over `Result<WithEncoded<OpTxEnvelope>>`
+    /// Helper: returns first successfully recovered transaction, if any.
+    pub fn try_into_recovered(&self) -> Option<Recovered<OpTxEnvelope>> {
+        self.try_to_recovered().find_map(Result::ok)
+    }
+
+    /// Returns iterator over decoded transactions with their original encoded bytes.
     pub fn decoded_transactions_with_encoded(
         &self,
-    ) -> impl Iterator<Item = Result<WithEncoded<OpTxEnvelope>, Box<dyn error::Error>>> + '_ {
+    ) -> impl Iterator<Item = Result<WithEncoded<OpTxEnvelope>, SignatureError>> + '_ {
         self.transactions.iter().flatten().map(|tx_bytes| {
             let env = OpTxEnvelope::decode_2718(&mut tx_bytes.as_ref())
-                .map_err(|e| Box::<dyn error::Error>::from(format!("Decode error: {e}")))?;
+                .map_err(|_| SignatureError::FromBytes("Failed to decode OpTxEnvelope"))?;
 
             let op_tx = OpTxEnvelope::try_from(env)
-                .map_err(|e| Box::<dyn error::Error>::from(format!("Conversion error: {e}")))?;
+                .map_err(|_| SignatureError::FromBytes("Failed to convert envelope"))?;
 
             Ok(WithEncoded::new(tx_bytes.clone(), op_tx))
         })
     }
-    /// Returns an iterator over successfully decoded `WithEncoded<Recovered<OpTxEnvelope>>` values.
-    ///
-    /// Transactions that fail to decode are silently filtered out.
+
+    /// Returns iterator over successfully recovered transactions with encoded bytes.
     pub fn recovered_transactions_with_encoded(
         &self,
-    ) -> impl Iterator<
-        Item = Result<WithEncoded<Recovered<OpTxEnvelope>>, Box<dyn core::error::Error>>,
-    > + '_ {
+    ) -> impl Iterator<Item = Result<WithEncoded<Recovered<OpTxEnvelope>>, SignatureError>> + '_
+    {
         self.transactions.iter().flatten().map(|tx_bytes| {
-            self.try_into_recovered(tx_bytes)
+            self.try_to_recovered()
+                .find_map(Result::ok)
                 .map(|recovered| WithEncoded::new(tx_bytes.clone(), recovered))
-                .ok_or_else(|| {
-                    Box::<dyn error::Error>::from(format!(
-                        "Recovery error: failed to decode transaction bytes: {:?}",
-                        tx_bytes
-                    ))
-                })
+                .ok_or(SignatureError::FromBytes(
+                    "Recovery error: failed to decode transaction bytes",
+                ))
```

Comment:
this one can reuse `decoded_transactions_with_encoded` and map the items accordinly